gcip/core/job.py

```diff
@@ -200,16 +200,16 @@ def _extend_name(self, name: Optional[str]) -> None:
         if name:
             self._name += "-" + name.replace("_", "-")
 
-    def _extend_stage(self, stage: Optional[str]) -> None:
+    def _extend_stage_value(self, stage: Optional[str]) -> None:
         """This method is used by `gcip.core.sequence.Sequence`s to populate the jobs stage."""
         if stage:
             self._stage += "_" + stage.replace("-", "_")
 
-    def _extend_namespace(self, namespace: Optional[str]) -> None:
+    def _extend_stage(self, stage: Optional[str]) -> None:
         """This method is used by `gcip.core.sequence.Sequence`s to populate the jobs name and stage."""
-        if namespace:
-            self._extend_name(namespace)
-            self._extend_stage(namespace)
+        if stage:
```

Comment:
Here lays the answere. 
`stage` is never uninitialised. Therefore `_extend_name` and `_extend_stage_value` shouldn't take `stage` as an optional argument. Saves some lines of  code.

Reply:
Look at the `populated_jobs` property of the `Sequence` class. There this method is called with optional namespace.

Reply:
OK